modules/http4s-server/src/main/scala/io/janstenpickle/trace4cats/http4s/server/Http4sContextConstructors.scala

```diff
@@ -0,0 +1,27 @@
+package io.janstenpickle.trace4cats.http4s.server
+
+import cats.Applicative
+import io.janstenpickle.trace4cats.Span
+import io.janstenpickle.trace4cats.http4s.common.{Http4sHeaders, Http4sRequestFilter, Http4sSpanNamer, Request_}
+import io.janstenpickle.trace4cats.inject.{ContextConstructor, SpanParams}
+import io.janstenpickle.trace4cats.model.SpanKind
+
+object Http4sContextConstructors {
+  def fromHeadersContext[F[_]: Applicative, Ctx](
+    makeContext: (Request_, Span[F]) => F[Ctx],
+    spanNamer: Http4sSpanNamer = Http4sSpanNamer.methodWithPath,
+    requestFilter: Http4sRequestFilter = Http4sRequestFilter.allowAll
+  )(cc: ContextConstructor[F, SpanParams, Span[F]]): ContextConstructor[F, Request_, Ctx] =
+    fromHeaders[F](spanNamer, requestFilter)(cc).mapBothF(makeContext)
```

Comment:
Thi will look as
```scala
fromHeaders[F](spanNamer, requestFilter)(cc).withInput.evalMap(makeContext.tupled)
```
A bit more verbose, but the set of operators is smaller and more "combinable".